tests/reducer-test.js

```diff
@@ -98,8 +102,124 @@ tape('test reducer case SELECT_GOAL: current goal is set to selected goal', (t)
+  t.equal(reducer(initialState, actionStepRateGoal).previousStep, steps.VIEW_GOAL, "step add goal sets correct step");
+  t.end();
+});
+
+tape('test reducer step_feedback: step and previousStep changed', (t) => {
+
+  const initialState = defaultState;
+  const actionStepFeedback = {
+    type: types.STEP_FEEDBACK,
+  };
+
+  t.equal(reducer(initialState, actionStepFeedback).step, steps.FEEDBACK, "step feedback sets correct step");
+  t.equal(reducer(initialState, actionStepFeedback).previousStep, steps.RATE_GOAL, "step feedback sets correct step");
+  t.end();
+});
+
+tape('test reducer MOVE_SLIDER: new rating added to currentGoal obj', (t) => {
+
+  const myGoal = { name: 'this is my goal', newRating: {}};
+  const initialState = {
+    ...defaultState,
+    goals: [ myGoal, ],
+    currentGoal: myGoal,
+  };
+  const newState = {
+    ...initialState,
+    currentGoal: {
+      ...initialState.currentGoal,
+      newRating: {score: 5},
+    }
+  };
+  const actionMoveSlider = {
+    type: types.MOVE_SLIDER,
+    rating: 5,
+  };
+
+  t.deepEqual(reducer(initialState, actionMoveSlider), newState, 'rating of 5 added to current state');
+  t.end();
+});
+
+tape('test reducer set pending sync open: pending sync set to open', (t) => {
+
+  const initialState = {
+    goals: [{id: 1, pendingSync: {open: false}}]
+  };
+  const newState = {
+    goals: [{id: 1, pendingSync: {open: true}}]
+  };
+  const actionSetPendingSyncOpen = {
+    type: types.SET_PENDING_SYNC_OPEN,
+    id: 1
+  };
+
+  t.deepEqual(reducer(initialState, actionSetPendingSyncOpen), newState, "Sets pending to true");
+  t.end();
+});
+
+tape('test reducer UPDATE_SYNC_SUCCESS: sync set to 1 and pending sync set to false', (t) => {
+
+  const initialState = {
+    goals: [{id: 1, syncDBCount: 0, pendingSync: {open: true}}]
+  };
+  const newState = {
+    goals: [{id: 1, syncDBCount: 1, pendingSync: {open: false}}]
+  };
+  const actionUpdateSyncSuccess = {
+    type: types.UPDATE_SYNC_SUCCESS,
+    id: 1
+  };
+
+  t.deepEqual(reducer(initialState, actionUpdateSyncSuccess), newState, "Sets pending to false and syncDBCount to 1");
+  t.end();
+});
+
+tape('test reducer UPDATE_SYNC_SUCCESS: pending sync set to false', (t) => {
+
+  const initialState = {
+    goals: [{id: 1, pendingSync: {open: true}}]
+  };
+  const newState = {
+    goals: [{id: 1, pendingSync: {open: false}}]
+  };
+  const actionUpdateSyncFailure = {
+    type: types.UPDATE_SYNC_FAILURE,
+    id: 1
+  };
+
+  t.deepEqual(reducer(initialState, actionUpdateSyncFailure), newState, "Sets pending to false");
+  t.end();
+});
+
+tape('test reducer RESET_UPDATE_COUNT: updateCount and syncDBCount set to 0', (t) => {
+
+  const initialState = {
+    goals: [{id: 1, updateCount: 1, syncDBCount: 2}]
+  };
+  const newState = {
+    goals: [{id: 1, updateCount:0, syncDBCount: 0}]
+  };
+  const actionResetUpdateCount = {
+    type: types.RESET_UPDATE_COUNT,
```

Comment:
Why do you want to do this? Surely you update count should always increase, if it goes down then bad things can happen?